packages/store/src/internal/validation/config-validator.ts

```diff
@@ -0,0 +1,32 @@
+import { Inject, Injectable } from '@angular/core';
+
+import {
+  CONFIG_MESSAGES as MESSAGES,
+  VALIDATION_CODE as CODE
+} from '../validation/validation-messages.config';
+import { NG_DEV_MODE, NGXS_TEST_MODE, NgxsConfig } from '../../symbols';
+
+@Injectable()
+export class ConfigValidator {
+  constructor(
+    @Inject(NGXS_TEST_MODE) public isTestMode: Function,
```

Comment:
Sorry, I didn't notice this before. This is based on the mode of Angular, not NGXS, so it should rather be called `NG_TEST_MODE`.
So we would be injecting `NG_TEST_MODE` and `NG_DEV_MODE` tokens.... hmmm... they seem to go together...
Thinking about it now, how about we have an class called `HostEnvironment` and we have `isTestMode` and `isDevMode` methods on it?

Reply:
good idea

Reply:
done